src/Foundation/NSArray.cs

```diff
@@ -63,7 +63,44 @@ public static NSArray FromNSObjects (int count, params INativeObject [] items)
 		{
 			return FromNativeObjects (items);
 		}
+		public static NSArray FromNSObjects<T> (params T [][] items) where T: class, INativeObject
+		{
+			if (items == null)
+				return null;
+			
+			var ret = new NSMutableArray ((nuint)items.Length);
+			foreach (var t in items) {
+				if (t == null)
+					throw new ArgumentNullException(nameof (t));
```

Comment:
Is the use of ```C# nameof()```correct here ?

Reply:
No, the caller does not know about local variables, so it does not make sense to use it in the exception message.

I would do something like this:

```csharp
for (var i = 0; i < items.Length; i++) {
    var row = items [i];
    if (row == null)
        throw new ArgumentNullException (nameof (items), $"Element [{i}] is null");
    for (var j = 0; j < row.Length; j++) {
        var element = row [j];
        if (element == null)
            throw new ArgumentNullException (nameof (items), $"Element [{i}][{j}] is null");
       ret.Add (NSArray.FromNSObjects (row));
    }
}
```